### DISCUSSION TITLE
`Session.is_modified` compares to previously flushed or committed value (not to just committed value as stated in the documentation)

### DISCUSSION THREAD

**Discussion by rdruzhkov** [sqlalchemy/sqlalchemy#11481] [Usage Questions]
In [the documentation of `Session.is_modified`](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.is_modified) method the following is stated:

> This method retrieves the history for each instrumented attribute on the instance and performs a comparison of the current value to its previously **committed** value, if any.

However, my experiments show that a comparison is done to its previously **flushed or committed** value.

Here is a minimal code example that should run as is:
```Python
from sqlalchemy import String
from sqlalchemy.orm import DeclarativeBase, Session
from sqlalchemy.orm import Mapped
from sqlalchemy.orm import mapped_column
from sqlalchemy import create_engine


class Base(DeclarativeBase):
    pass


class User(Base):
    __tablename__ = "user_account"
    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str] = mapped_column(String(30))


if __name__ == '__main__':
    engine = create_engine("sqlite://")
    Base.metadata.create_all(engine)

    with Session(engine) as session:
        user = User(id=1, name="John")
        session.add(user)
        
        user.name = "Arthur"  # Modify the instance
        
        print(f"Before flush: session.is_modified(user) = {session.is_modified(user)}")
        session.flush()
        
        print(f"After flush, before commit: session.is_modified(user) = {session.is_modified(user)}")
        
        session.commit()
        print(f"After commit: session.is_modified(user) = {session.is_modified(user)}")
```

The output is the following:
```
Before flush: session.is_modified(user) = True
After flush, before commit: session.is_modified(user) = False
After commit: session.is_modified(user) = False
```

I believe it would be better to state that comparison is done to committed or flushed value. Otherwise, it may confuse what event listener to use (`before_flush` or `before_commit`) for tracking instance changes.

What do you think?

**Comment by CaselIT** [comment]
Hi,

Makes sense to improve the docs. Can you create a PR?

**Reply by rdruzhkov** [reply]
Yeah, sure. I will create a PR within a few days.

**Reply by rdruzhkov** [reply]
@CaselIT I've created a PR (https://github.com/sqlalchemy/sqlalchemy/pull/11501), please, take a look.

**Reply by CaselIT** [reply]
Thanks!
